yacron/config.py

```diff
@@ -316,6 +325,44 @@ def __init__(self, config: dict) -> None:
                     "but yacron is not running as superuser".format(self.name)
                 )
 
+    def __parse_environs(self):
+        """
+        Parse env_file and environments configuration parameters.
+
+        Opens `env_file` and loads variables, then merges them with the `environment` ones.
+        The latter will replace the former if duplicated.
+        """
+        # a temporary dictionary to prevent duplicated keys.
+        # why are you using a list?
+        environ: Dict[str, str] = {}
+
+        # if `env_file` is specified, load the file contents
+        if self.env_file is not None:
+            with open(self.env_file, 'r') as env_file:
```

Comment:
I would move this part of the code (until line 357) into a separate utility function.  Something like:
`def read_env_file(file_name: str) -> Dict[str, str]:`

Reply:
that's acknowledged, I made a  method to avoid writing even more code in the constructor (even with helpers, it's required to merge the environs together)